common/src/logging.rs

```diff
@@ -100,8 +100,46 @@ macro_rules! log_if_error_fmt {
     }};
 }
 
+pub struct Hidden<T> {
+    inner: T,
+}
+
+impl<T> Hidden<T> {
+    pub fn new(inner: T) -> Self {
+        Self { inner }
+    }
+}
+
+impl<T> fmt::Debug for Hidden<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "[FILTERED]")
+    }
+}
+
+impl<T> fmt::Display for Hidden<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "[FILTERED]")
+    }
+}
+
+impl<T> std::ops::Deref for Hidden<T> {
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl<T> DerefMut for Hidden<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.inner
+    }
+}
+
```

Comment:
Looks great! I would only add implementation `From<T> for Hidden<T>` . So you can go `32.into()` but up to you.

I think we should add this to the tari_utilities crate. This can also be included here and refactored out once tari_utilities has been released.

Reply:
+1 for the `From<T>` implementation. Also you can add `into_inner(self) -> T` method to take the data out.